base/compiler/abstractinterpretation.jl

```diff
@@ -1071,6 +1071,7 @@ function abstract_eval_value_expr(interp::AbstractInterpreter, e::Expr, vtypes::
         if 1 <= n <= length(sv.sptypes)
             t = sv.sptypes[n]
         end
+        return t
```

Comment:
Currrently `abstract_eval_value_expr` will return `nothing` when the `if` condition above is `false`; I guess this function is better to return types instead of that, right ? Not sure what will actually happen when this returns `nothing` though.

Reply:
Whoops, thanks!